src/Symfony/Component/Mime/NamedAddress.php

```diff
@@ -16,6 +16,8 @@
  */
 final class NamedAddress extends Address
 {
+    private const FROM_STRING_PATTERN = '~(?<name>.*)<(?<address>[^>]+)>~';
```

Comment:
We should have a look at RFC 2822 to be sure we cover most common use cases.

Reply:
I think I'm going to need some help here. As far as I understand from https://tools.ietf.org/html/rfc2822#section-3.4 the `angle-addr` is simply matched with this regex, but `addr-spec` might contain a < which is not checked in this regex (as in 'Me <"name-with<address"@domain.com>) which is not 'common' I think. Any more checking should be done while construction the (Named)Address, which is already done.

Reply:
You are a brave soul. Welcome to email addresses 😄 funny enough you can use any character in the email address (and name part) as long as it's quoted, and better yet you can even use quotes in quotes, as long as those are properly escaped! I believe you can even use `@` in the name part 😐

Parsing a mailbox header with a regex is like parsing HTML with a regex, possible but you are gonna miss a lot edge-cases.

During the Fossa hackaton while struggling to come-up with a way to parse an email message (and its headers) I came discover that you can only properly parse a MIME header using a full fledged lexer, as there are so many edge cases. _But, luckily I managed to solve my use-case another way 😊 -- that didn't involve a complete message parser._

If we don't want to support all the most complex cases, it's properly best to look for the first and last `<>` and assume this as the actual address, and use everything before the first `<` as the name. That is safest and fastest.

Reply:
I haven't checked, but can't we have comments as well? We can probably don't implement them though. If we keep the simple regex, let's at least add a comment to describe what we are doing and what we don't want to support.

Reply:
Ah yes comments 😭 

> If we keep the simple regex, let's at least add a comment to describe what we are doing and what we don't want to support.

👍 

Plus. Anyone who must use a complex name plus address can still construct the object manually.

Reply:
> it's properly best to look for the first and last <> and assume this as the actual address, and use everything before the first < as the name. That is safest and fastest.

I'll update the regex to match the first `<` to the last `>`.


> If we keep the simple regex, let's at least add a comment to describe what we are doing and what we don't want to support.

I'll add that comment

Reply:
Thanks @sstok for helping me. I updated the regex and added a comment